code/datums/traits/family_heirloom.dm

```diff
@@ -0,0 +1,912 @@
+			/obj/item/shovel)
+	if(lencheck)
+		return length(heirloom_items_base)
+	return pick(heirloom_items_base)
+
+//COMMANDERS
+/datum/job/captain/get_heirloom_list(var/lencheck)
+	var/static/list/heirloom_items_base
+	if(!heirloom_items_base)
+		heirloom_items_base = list(
+			/obj/item/reagent_containers/food/drinks/flask/gold)
+	if(lencheck)
+		return 4 //hich chance of spawning captain's flask
+	return pick(heirloom_items_base)
+
+// head of personnel
+/datum/job/hop/get_heirloom_list(var/lencheck)
+	var/static/list/heirloom_items_base
+	if(!heirloom_items_base)
+		heirloom_items_base = list(
+			/obj/item/toy/plush/ian)
+	if(lencheck)
+		return length(heirloom_items_base)
+	return pick(heirloom_items_base)
+
+
+//------------------DEPARTMENT ROLLS SET UP-----------------------
+/datum/quirk/family_heirloom/proc/get_department_items(var/lencheck, var/mob/living/carbon/human/H)
+	var/static/DEPARTMENT_FLAG // Bitflag for department shared items
+	var/tablesize = 0
+
+	if(lencheck)
+		DEPARTMENT_FLAG = 0 //We're going to initialize this when we calculate size
+
+	// This IF checks your job then turn on the bitflag
+	// It's to get multiple departments from a mob rather than to get their 'real department'
+	// "|| prob(1)" means you get this flag at 1% chance
+	//SERVICE
+	if(H.mind.assigned_role in list(
+		    JOB_ASSISTANT,
+		    JOB_JANITOR,
+		    JOB_BARTENDER,
+		    JOB_COOK,
+		    JOB_BOTANIST,
+		    JOB_CURATOR,
+		    JOB_CHAPLAIN,
+		    JOB_LAWYER,
+		    JOB_BARBER,
+		    JOB_DEBTOR,
+		    JOB_CLOWN,
+		    JOB_MIME,
+		    JOB_STAGE_MAGICIAN,
+			JOB_PSYCHIATRIST, //this is the true nature of your job.
+		    JOB_VIP
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_SERVICE
+
+	//SECURITY
+	if(H.mind.assigned_role in list(
+		    JOB_HOS,
+		    JOB_WARDEN,
+			JOB_SECURITY_OFFICER,
+			JOB_DETECTIVE,
+			JOB_DEPUTY,
+			JOB_BRIG_PHYSICIAN
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_SECURITY
+
+	//SCIENCE
+	if(H.mind.assigned_role in list(
+		    JOB_RD,
+		    JOB_SCIENTIST,
+			JOB_EXPLORATION_CREW,
+			JOB_ROBOTICIST
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_SCIENCE
+
+	//MEDICAL
+	if(H.mind.assigned_role in list(
+		    JOB_CMO,
+		    JOB_BRIG_PHYSICIAN,
+			JOB_MEDICAL_DOCTOR,
+			JOB_PARAMEDIC,
+			JOB_PSYCHIATRIST,
+			JOB_CHEMIST,
+			JOB_VIROLOGIST,
+			JOB_GENETICIST
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_MEDICAL
+
+	//ENGINEERING
+	if(H.mind.assigned_role in list(
+		    JOB_CE,
+			JOB_STATION_ENGINEER,
+			JOB_ATMOSPHERIC_TECHNICIAN
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_ENGINEERING
+
+	//CARGO
+	if(H.mind.assigned_role in list(
+		    JOB_QM,
+		    JOB_CARGO_TECHNICIAN,
+			JOB_SHAFT_MINER
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_CARGO
+
+	//COMMAND
+	if(H.mind.assigned_role in list(
+			JOB_CAPTAIN,
+			JOB_HOP,
+		    JOB_HOS,
+			JOB_RD,
+			JOB_CMO,
+			JOB_CE,
+			JOB_VIP // just for giving annoying items
+		) || prob(1))
+		DEPARTMENT_FLAG |= DEPARTMENT_COMMAND
```

Comment:
Don't jobs store what department they are a part of? This list seems dodgy to me since its another thing you have to update if you add a new job that people will forget.